modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java

```diff
@@ -115,7 +116,14 @@
         int status = e instanceof IgniteClientException ?
             ((IgniteClientException)e).statusCode() : ClientStatus.FAILED;
 
-        return new ClientResponse(req.requestId(), status, e.getMessage());
+        boolean fullStack = false;
+
+        if (X.getCause(e) != null || !X.getSuppressedList(e).isEmpty()) {
+            fullStack =
+                ctx.kernalContext().config().getClientConnectorConfiguration().getThinClientConfiguration().showFullStack();
```

Comment:
Why don't you want to do this in constructor?
I see you use static configuration as a dynamic, but I think it doesn't right too.

Reply:
cause it can be changed through jmx.

Reply:
I know it, but I do not recommend to change static configuration.
Can you add a specific volatile flag in the ClientRequestHandler and update directly it thought JMX?